File: stripe_terminal/ios/Classes/Mappings/RefundMappings.swift
import Foundation
import StripeTerminal

extension Refund {
    func toApi() -> RefundApi {
        return RefundApi(
            amount: Int(amount),
            chargeId: charge,
            created: created,
            currency: currency,
            failureReason: failureReason,
            id: stripeId,
            metadata: metadata,
            paymentMethodDetails: paymentMethodDetails?.toApi(),
            reason: reason,
            status: status.toApi()
        )
    }
}

extension RefundStatus {
    func toApi() -> RefundStatusApi? {
        switch self {
        case .succeeded:
            return .succeeded
        case .pending:
            return .pending
        case .failed:
            return .failed
        case .unknown:
            return nil
        @unknown default:
            fatalError()
        }
    }
}

extension PaymentMethodDetails {
    func toApi() -> PaymentMethodDetailsApi {
        return PaymentMethodDetailsApi(
            cardPresent: cardPresent?.toApi(),
            interacPresent: interacPresent?.toApi()
        )
    }
}

File: stripe_terminal/android/src/main/kotlin/mek/stripeterminal/mappings/RefundMappings.kt
package mek.stripeterminal.mappings

import com.stripe.stripeterminal.external.models.Refund
import mek.stripeterminal.api.RefundApi
import mek.stripeterminal.api.RefundStatusApi
import mek.stripeterminal.toHashMap

fun Refund.toApi(): RefundApi {
    return RefundApi(
        id = id,
        amount = amount!!,
        chargeId = chargeId!!,
        created = created!!,
        currency = currency!!,
        metadata = metadata?.toHashMap() ?: HashMap(),
        reason = reason,
        status = when (status) {
            "succeeded" -> RefundStatusApi.SUCCEEDED
            "pending" -> RefundStatusApi.PENDING
            "failed" -> RefundStatusApi.FAILED
            else -> null
        },
        paymentMethodDetails = paymentMethodDetails?.toApi(),
        failureReason = failureReason
    )
}
